zcash_client_backend/src/fees.rs

```diff
@@ -368,13 +443,30 @@ impl EphemeralBalance {
 /// A trait that represents the ability to compute the suggested change and fees that must be paid
 /// by a transaction having a specified set of inputs and outputs.
 pub trait ChangeStrategy {
-    type FeeRule: FeeRule;
+    type FeeRule: FeeRule + Clone;
     type Error;
 
+    /// The type of metadata source that this change strategy requires in order to be able to
+    /// retrieve required wallet metadata. If more capabilities are required of the backend than
+    /// are exposed in the [`InputSource`] trait, the implementer of this trait should define their
+    /// own trait that descends from [`InputSource`] and adds the required capabilities there, and
+    /// then implement that trait for their desired database backend.
+    type MetaSource: InputSource;
+    type WalletMeta;
```

Comment:
* Please document this type.
```suggestion
    type WalletMetaT;
```